src/api/Nhs.Appointments.Api/Models/GetAvailabilityCreatedEventsRequest.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using Newtonsoft.Json;
+
+namespace Nhs.Appointments.Api.Models;
+public record GetAvailabilityCreatedEventsRequest(
+    [JsonProperty("site")]
+    string Site,
+    [JsonProperty("from")]
+    string From
+)
+
+{
+    public DateOnly FromDate => DateOnly.ParseExact(From, "yyyy-MM-dd");
```

Comment:
You don't need to follow this pattern any more. You can (and should) use DateOnly on the json property.